pkg/execution/executor/executor.go

```diff
@@ -432,8 +432,11 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 
 	// This is a success, which means either a generator or a function result.
 	if len(resp.Generator) > 0 {
+		copiedItem := item
+		copiedItem.DisableImmediateExection = len(resp.Generator) > 1
+
 		// Handle generator responses then return.
-		if serr := e.HandleGeneratorResponse(ctx, resp.Generator, item); serr != nil {
+		if serr := e.HandleGeneratorResponse(ctx, resp.Generator, copiedItem); serr != nil {
```

Comment:
This feels pretty dirty, though passing more options to every function down the line also feels crap.